### ISSUE TITLE
Vulkan Validation Error with draw_indirect

### ISSUE THREAD

**Issue by almarklein** [gfx-rs/wgpu#1603]
In one of our test suite we run a series of tests with the following vertex shader:

```wgsl
[[stage(vertex)]]
fn vs_main([[builtin(vertex_index)]] vertex_index : u32) -> [[builtin(position)]] vec4<f32> {
    var positions: array<vec3<f32>, 4> = array<vec3<f32>, 4>(
        vec3<f32>(-0.5, -0.5, 0.1),
        vec3<f32>(-0.5,  0.5, 0.1),
        vec3<f32>( 0.5, -0.5, 0.1),
        vec3<f32>( 0.5,  0.5, 0.1),
    );
    let p: vec3<f32> = positions[vertex_index];
    return vec4<f32>(p, 1.0);
}
```

Since the latest bump of wgpu-core in wgpu-native, this started producing a validation error, for the cases where we use `render_pass.draw_indirect()` and `render_pass.draw_indexed_indirect()`:

```
VALIDATION [VUID-vkCmdPipelineBarrier-dstAccessMask-02816 (0x69c8467d)]
	Validation Error: [ VUID-vkCmdPipelineBarrier-dstAccessMask-02816 ] Object 0: handle = 0x24d7c519648, name = _Transit, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x69c8467d | vkCmdPipelineBarrier(): .pMemoryBarriers[0].dstAccessMask bit VK_ACCESS_INDIRECT_COMMAND_READ_BIT is not supported by stage mask (VK_PIPELINE_STAGE_VERTEX_INPUT_BIT|VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT). The Vulkan spec states: The dstAccessMask member of each element of pMemoryBarriers must only include access flags that are supported by one or more of the pipeline stages in dstStageMask, as specified in the table of supported access types (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdPipelineBarrier-dstAccessMask-02816)
	objects: (type: COMMAND_BUFFER, hndl: 0x24d7c519648, name: _Transit)
```

This is on Windows 10.

**Comment by almarklein** [comment]
A very similar validation error occurs in another test with a compute shader and `dispatch_indirect()`:

```wgsl
        [[block]]
        struct DataContainer { data: [[stride(4)]] array<i32>; };

        [[group(0), binding(0)]]
        var<storage> data1: [[access(read)]] DataContainer;

        [[group(0), binding(1)]]
        var<storage> data2: [[access(write)]] DataContainer;

        [[stage(compute), workgroup_size(1)]]
        fn main([[builtin(global_invocation_id)]] index: vec3<u32>) {
            let i = i32(index.x);
            data2.data[i] = data1.data[i] + 1;
        }
```

```
VALIDATION [VUID-vkCmdPipelineBarrier-dstAccessMask-02816 (0x69c8467d)]
	Validation Error: [ VUID-vkCmdPipelineBarrier-dstAccessMask-02816 ] Object 0: handle = 0x207df0af878, name = _Transit, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x69c8467d | vkCmdPipelineBarrier(): .pMemoryBarriers[2].dstAccessMask bit VK_ACCESS_INDIRECT_COMMAND_READ_BIT is not supported by stage mask (VK_PIPELINE_STAGE_VERTEX_INPUT_BIT|VK_PIPELINE_STAGE_VERTEX_SHADER_BIT|VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT|VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT|VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT). The Vulkan spec states: The dstAccessMask member of each element of pMemoryBarriers must only include access flags that are supported by one or more of the pipeline stages in dstStageMask, as specified in the table of supported access types (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdPipelineBarrier-dstAccessMask-02816)
	objects: (type: COMMAND_BUFFER, hndl: 0x207df0af878, name: _Transit)
```

**Comment by cwfitzgerald** [comment]
What version of wgpu was this with?

**Comment by kvark** [comment]
Sorry about this! fixed in 2efccb2bf83afa9f1ef235e97719ec50da3128a4
